### PR TITLE
[engraving] Added LayoutData... Step 49

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: src/engraving/rendering/dev/tlayout.cpp [hunk 1]
```diff
@@ -2205,14 +2205,14 @@ void TLayout::layout(FretDiagram* item, LayoutContext& ctx)
...
 
-    if (!item->explicitParent() || !item->explicitParent()->isSegment()) {
-        item->setPos(PointF());
+    if (!item->explicitParent()->isSegment()) {
+        ldata->setPos(PointF());
         return;
     }
 
-    // We need to get the width of the notehead/rest in order to position the fret diagram correctly
+    // We need to get th)e width of the notehead/rest in order to position the fret diagram correctly
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/tlayout.cpp:2215]: Unnecessary change :)

File: src/engraving/rendering/dev/tlayout.cpp [hunk 2]
```diff
@@ -2301,10 +2301,15 @@ void TLayout::layout(FretCircle* item, LayoutContext&)
     double offsetFromUpNote = (item->rect().height() - chordRect.height()
                                - (item->chord()->downNote()->pos().y() - item->chord()->upNote()->pos().y())
                                ) / 2;
-    item->setOffsetFromUpNote(offsetFromUpNote);
-    item->setSideOffset((item->rect().width() - minWidth) / 2);
+    ldata->offsetFromUpNote = offsetFromUpNote;
+    ldata->sideOffset = ((item->rect().width() - minWidth) / 2);
+
+    ldata->setbbox(item->rect().adjusted(-lw, -lw, lw, lw));
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/tlayout.cpp]: What do you think about replacing `item->rect()` with `ldata->rect` in these lines?

File: src/engraving/rendering/dev/arpeggiolayout.cpp [hunk 3]
```diff
@@ -151,6 +151,8 @@ void ArpeggioLayout::layoutOnEdit(Arpeggio* item, LayoutContext& ctx)
 //---------------------------------------------------------
 void ArpeggioLayout::symbolLine(const IEngravingFontPtr& f, Arpeggio::LayoutData* data, SymId end, SymId fill)
 {
+    data->symbols.clear();
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/arpeggiolayout.cpp:154]: Looks like this will resolve #19083, right?

File: src/engraving/rendering/dev/tlayout.cpp [hunk 4]
```diff
@@ -2312,33 +2312,24 @@ void TLayout::layout(FretCircle* item, LayoutContext& ctx)
...
-    Note* anchor2     = toNote(item->endElement());
-    Chord* cr1         = anchor1->chord();
-    Chord* cr2         = anchor2->chord();
-    GlissandoSegment* segm1 = toGlissandoSegment(item->frontSegment());
-    GlissandoSegment* segm2 = toGlissandoSegment(item->backSegment());
+    Note* anchor1 = toNote(item->startElement());
+    Note* anchor2 = toNote(item->endElement());
+    Chord* cr1 = anchor1->chord();
+    Chord* cr2 = anchor2->chord();
+    GlissandoSegment* segm1 = toGlissandoSegment(const_cast<Glissando*>(item)->frontSegment());
```
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/tlayout.cpp:2331]: This is getting a little bit awful with all those const_cast... and it brings us to the difficult question of what to do with spanner segments. We can leave it like this for now, or we can make the `Glissando*` not `const`; both are not ideal of course, but it's temporary anyway.

**Review by cbjeukendrup — APPROVED** [review]

File: src/engraving/rendering/dev/arpeggiolayout.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/engraving/rendering/dev/arpeggiolayout.cpp:154]: yes (

File: src/engraving/rendering/dev/tlayout.cpp (hunk 4 above)
Comment (igorkorsukov) [reply src/engraving/rendering/dev/tlayout.cpp:2331]: At the moment `const_cast` indicate problems places, this is hack. We should think about each `const_cast`, but I think will be next iteration of work, because now it just technikaly changes, without changes algoritm. I would not like mix tech and algo changes.

File: src/engraving/rendering/dev/tlayout.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/engraving/rendering/dev/tlayout.cpp:2215]: oops )
